src/components/ReportingConfig/index.jsx

```diff
@@ -0,0 +1,175 @@
+  }
+
+  /**
+   * Creates a new reporting configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  createConfig = async (formData) => {
+    //  snake_case the data before sending it to the backend
+    const transformedData = snakeCaseFormData(formData);
+    try {
+      const response = await LMSApiService.postNewReportingConfig(transformedData);
+      this.setState(prevState => ({
+        reportingConfigs: [
+          ...prevState.reportingConfigs,
+          response.data,
+        ],
+      }));
+      this.newConfigFormRef.current.close();
+      return undefined;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  deleteConfig = async (uuid) => {
+    try {
+      const response = await LMSApiService.deleteReportingConfig(uuid);
+      const deletedIndex = this.state.reportingConfigs
+        .findIndex(config => config.uuid === response.data.uuid);
+      const reportingConfigsCopy = this.state.reportingConfigs;
+      reportingConfigsCopy.splice(deletedIndex, 1);
+      this.setState({
+        reportingConfigs: reportingConfigsCopy,
+      });
+      return undefined;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+   * Updates an existing reporting configuration. Returns if there is an
+   * error.
+   * @param {FormData} formData
+   * @param {string} uuid -> The uuid of the configuration we are updating
+   */
+  updateConfig = async (formData, uuid) => {
+    const transformedData = snakeCaseFormData(formData);
+    try {
+      const response = await LMSApiService.updateReportingConfig(transformedData, uuid);
+      const updatedIndex = this.state.reportingConfigs
+        .findIndex(config => config.uuid === response.data.uuid);
+      const reportingConfigsCopy = [...this.state.reportingConfigs];
+      reportingConfigsCopy[updatedIndex] = response.data;
+      this.setState({ reportingConfigs: [...reportingConfigsCopy] });
+      return undefined;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  render() {
+    const { reportingConfigs, loading, error } = this.state;
+
+    if (loading) {
+      return <LoadingMessage className="overview" />;
+    }
+
+    if (error) {
+      return <ErrorPage />;
+    }
+    return (
+      <div>
+        <React.Fragment>
+          {reportingConfigs &&
+            reportingConfigs.map(config => (
+              <div
+                key={config.uuid}
+                className="mb-3"
+              >
+                <Collapsible
+                  styling="card"
+                  className="shadow"
+                  title={
+                    <div className="row justify-content-around">
+                      <Icon
+                        className={`col-1 ${config.active ? ' fa fa-check text-success-300' : ' fa fa-times text-danger-300'}`}
+                      />
+                      <div className="col">
+                        <h6>Report Type:</h6>
```

Comment:
Are we actually at the 6th header level, or was this done for style? Make sure we aren't skipping levels and causing a11y issues. It looks like we don't get below `h3` anywhere in the admin portal.

Reply:
I've changed these to h3 with the h6 class. If there's a better, more a11y way to do this, I'd be happy to change it.